app/grandchallenge/discussion_forums/templates/discussion_forums/forumpost_list.html

```diff
@@ -1,46 +1,90 @@
 {% extends "discussion_forums/forum_base.html" %}
-{% load guardian_tags %}
 
+{% load static %}
+{% load guardian_tags %}
 
-{% block sub_title %}{{ topic }} - {% endblock %}
+{% block sub_title %}{% if topic %}{{ topic }}{% endif %}My posts- {% endblock %}
 
 {% block extra_breadcrumbs %}
     <li class="breadcrumb-item">
         <a href="{% url 'discussion-forums:topic-list' %}">Forum</a>
     </li>
-    <li class="breadcrumb-item active">{{ topic }}</li>
+    {% if topic %}
+        <li class="breadcrumb-item active">{{ topic }}</li>
+    {% else %}
+        <li class="breadcrumb-item active">My posts</li>
+    {% endif %}
 {% endblock %}
 
 {% block extra_top_row_buttons %}
     <div class="col-6">
-        <a href="{% url 'discussion-forums:topic-list' %}" class="btn btn-dark btn-sm">Back to forum</a>
+        <a href="{% url 'discussion-forums:topic-list' %}" class="btn btn-dark btn-sm"><i class="fas fa-arrow-left mr-1"></i>Back to forum</a>
+        <a href="{% url 'discussion-forums:my-posts' %}" class="btn btn-dark btn-sm"><i class="fas fa-comments mr-1"></i>My posts</a>
     </div>
 {% endblock %}
 
 {% block extra_content %}
-    <div class="row">
-        <div class="col-12"><h2>{{ topic.subject }}</h2></div>
-    </div>
-    {% get_obj_perms request.user for topic as "user_perms" %}
-    <div>
-        {% if 'create_topic_post' in user_perms %}
-            <a hx-get="{% url 'discussion-forums:post-create' slug=topic.slug %}" hx-swap="beforeend show:bottom" hx-target="#post-list" class="btn btn-primary btn-sm">
-                <i class="fas fa-comment mr-1"></i> Reply
-            </a>
-        {% endif %}
-        {% include 'actstream/partials/follow_unfollow_links.html' with object=topic %}
-        {% if 'delete_forumtopic' in  user_perms %}
-            <a href="{% url 'discussion-forums:topic-delete' slug=topic.slug %}" class="btn btn-danger btn-sm">
-                <i class="fas fa-trash mr-1"></i>Delete topic
-            </a>
-        {% endif %}
-    </div>
-    {# TODO: lock topic button #}
+    {% if topic %}
+        <div class="h3 my-3" title="{% if topic.is_locked %}This topic is locked.{% endif %}">
+            {% if topic.is_locked %}<i class="fas fa-lock mr-2 fa-xs badge-dark p-2 rounded"></i>{% endif %}
+            {{ topic.subject }}
+        </div>
+    {% else %}
+        <h3>My posts</h3>
+    {% endif %}
+
+    {% if topic %}
+        {% get_obj_perms request.user for topic as "user_perms" %}
+        <div class="row">
+            <div class="col-8">
+                {% if 'create_topic_post' in user_perms %}
+                    <div class="d-inline" title="{% if topic.is_locked %}This topic is locked. You cannot reply to it.{% endif %}">
+                        <a hx-get="{% url 'discussion-forums:post-create' slug=topic.slug %}" hx-swap="beforeend show:bottom" hx-target="#post-list" class="btn btn-primary btn-sm {% if topic.is_locked %}disabled{% endif %}">
```

Comment:
This is just disabling the button, the user can still post data and create a new post. Should the `create_topic_post` permission be removed and added when locking and unlocking the post, or it should be checked on PostCreate perhaps by adding `is_locked=False` in `get_object_or_404`. Both would need a test.

Reply:
There is validation in the form, so the user can post but it wouldn't create a post when the topic is locked. There is a test for it. I thought that would be enough?

Reply:
Ah missed that, sounds good!

Reply:
I guess I can also check whether the user is an admin in the view, of course, so I could use the `AccessMixin` to add the check for whether it's unlocked or the user is an admin.

Reply:
But so long as the user cannot make a post for a locked topic then that is fine.

Reply:
I'm wondering if it makes sense to allow challenge admins to post to locked topics. We could simplify things and just not allow posting to locked topics for anyone. The challenge admin can always unlock the topic again, post and lock it again if they really need to post to the topic. Then the check in the form could go and we could revoke `lock_forumtopic` for everyone when a topic gets locked. This also simplifies the templates, because else we need to check in the template if the user is a challenge admin in multiple place to decide whether to render the form and disable the buttons etc.

Reply:
I have implemented the permission revoking / regranting now. I like that solution best, I think. It also simplifies the templates: the reply button just won't appear for locked topics.